app/controllers/excursions_controller.rb

```diff
@@ -0,0 +1,40 @@
+class ExcursionsController < ApplicationController
+  before_action :setup_list, only: [:show, :update, :destroy]
+  def index
+    @excursions = Excursion.all
+  end
+
+  def show
+  end
+
+  def new
+    @excursion = Excursion.new
+  end
+
+  def create
+    @excursion = Excursion.new(excursion_params)
+    @excursion.save
+  end
+
+  def edit
+    @excursion = Excursion.find(:id)
```

Comment:
This line is in the private method in line 38,  so i would put the edit in the before action aswell and delete this line